src/group_theory/submonoid/membership.lean

```diff
@@ -234,8 +236,8 @@ lemma pow_right_injective_iff_pow_injective {n : M} :
   function.injective (λ m : ℕ, n ^ m) ↔ function.injective (pow n) :=
 subtype.coe_injective.of_comp_iff (pow n)
 
-theorem log_pow_eq_self [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m)) (m : ℕ) :
-  log (pow n m) = m :=
+@[simp] theorem log_pow_eq_self [decidable_eq M] {n : M} (h : function.injective (λ m : ℕ, n ^ m))
```

Comment:
Do we have `simp` lemmas that prove `h`?

Reply:
Otherwise LGTM
bors d+

Reply:
`h` isn't always true though. The closest we have are `int.pow_right_injective` and `nat.pow_right_injective`.